ops/pebble.py

```diff
@@ -1062,6 +1062,74 @@ def to_dict(self) -> 'CheckDict':
         dct = {name: value for name, value in fields if value}
         return typing.cast('CheckDict', dct)
 
+    def _merge(self, other: 'Check'):
+        """Merges this check object with another check definition.
+
+        For attributes present in both objects, the passed in check
+        attributes take precedence.
+        """
+        if other.level != '':
+            self.level = other.level
+        if other.period != '':
+            self.period = other.period
+        if other.timeout != '':
+            self.timeout = other.timeout
+        if other.threshold is not None:
+            self.threshold = other.threshold
+        if other.http is not None:
+            if other_url := other.http.get('url'):
+                if self.http is None:
+                    self.http = {}
+                self.http['url'] = other_url
+            if other_headers := other.http.get('headers'):
+                if self.http is None:
+                    self.http = {'headers': {}}
+                for header, value in other_headers.items():
+                    self.http['headers'][header] = value
```

Comment:
There's got to be a more elegant way to write this...

```suggestion
                self.http = {**(self.http or {}).get('headers', {}), **other_headers}
```

Another option, a little less magical:

```suggestion
                if self.http is None:
                    self.http = {'headers': {}}
                self.http["headers"].update(other_headers)
```

As a team, we should actually think about HTTP request header field names, because current API hides the fact that they are case-insensitive by RFC.

Meanwhile a naive merge of `{Host:...} + {host:...}` will end up with two copies differing by case.

@tonyandrewmeyer wdyt?

Reply:
I agree there's got to be a more elegant / lower-code way to write this. For example, if you look at the existing [`Service._merge`](https://github.com/canonical/operator/blob/d8148ab41869f6119683cf42b7e6e574f1a99bf1/ops/pebble.py#L992), it uses setattr/getattr, but special cases lists and dicts. This makes it significantly less code than [the Pebble version](https://github.com/canonical/pebble/blob/b45911ac73ca1991cdd1a1085e037a5ef9ed3e0c/internals/plan/plan.go#L122).

I think we can take the same approach for `LogTarget._merge`, and perhaps `Check._merge`, though the latter has the complication of the nested dicts that we'll need to figure out. Still, it seems to me we can simplify this.

Regarding @dimaqq's point about case sensitivity, I think that's fine as is. Pebble doesn't try to do anything clever here, so we shouldn't either.

Reply:
One more concern: is there a way to reset some field?

If a layer specifies `http.header.Authorization`, can a subsequent layer ever reset auth?